src/expm.jl

```diff
@@ -0,0 +1,64 @@
+@inline expm(A::StaticMatrix) = _expm(Size(A), A)
+
+@generated function _expm(::Size{(1,1)}, A::StaticMatrix)
+    @assert size(A) == (1,1)
+    T = promote_type(typeof(sqrt(one(eltype(A)))), Float32)
```

Comment:
In Julia v0.6, you can only perform "hyper"-pure operations inside the function generator. Since the eltype might not be defined before static arrays is loaded, this just isn't "pure enough" for the assumptions made by the Julia 0.6 compiler.

However, it's safe to move this into the body of the function (inside the `quote`) - where possible inference will perform type calculations at compile time.

Also - why `sqrt(one)` not `exp(zero)`? (zero is slightly more flexible from a group vs. algebra perspective, i.e. in theory `eltype(A)` might be a generator that supports addition and `zero` only.)